packages/sx.js/test/integration/offchain/index.test.ts

```diff
@@ -0,0 +1,33 @@
+import { Wallet } from '@ethersproject/wallet';
+import { EthereumSig } from '../../../src/clients/offchain/ethereum-sig';
+import { offchainGoerli } from '../../../src/offchainNetworks';
+import vote from './fixtures/vote.json';
+
+// Test address: 0xf1f09AdC06aAB740AA16004D62Dbd89484d3Be90
+const TEST_PK = 'ef4bcf36b5d026b703b86a311031fe2291b979620f01443f795fa213f9105e35';
+const signer = new Wallet(TEST_PK);
+const client = new EthereumSig({ networkConfig: offchainGoerli });
+
+describe('vote', () => {
+  describe('when valid', () => {
+    it('should vote', async () => {
+      const envelope = await client.vote({
+        signer,
+        data: vote
+      });
+
+      return expect(client.send(envelope)).resolves.toHaveProperty('id');
+    });
+  });
+
+  describe('when invalid', () => {
+    it('should thrown an error', async () => {
+      const envelope = await client.vote({
+        signer,
+        data: { ...vote, proposal: 'unknown-proposal' }
+      });
+
+      return expect(client.send(envelope)).rejects.toThrowError(/unknown proposal/);
+    });
+  });
```

Comment:
Nit, I wouldn't add extra `describe` for those cases as we now have just single `it` in each `describe` so it's a bit much, I'd just stick to `it should cast a vote` and `it should throw an error when proposal is invalid`.